services/net/filemonitor/FilemonitorAction.cs

```diff
@@ -386,22 +528,27 @@ private List<SourceContent> GetFmsArticles(string dir, IIngestServiceActionManag
                     var filtered = preFiltered.Replace("\n\n", Fields.FmsFieldDelim);
 
                     var papername = GetFmsData(filtered, Fields.Papername, ingest);
-                    source = string.IsNullOrEmpty(source) ? GetItemSourceCode(ingest, papername) : source;
-                    item.Source = source;
-                    item.Title = GetFmsData(filtered, Fields.Headline, ingest);
-                    item.Uid = GetFmsData(filtered, Fields.Id, ingest);
-                    item.Body = GetFmsData(preFiltered + "<break>", Fields.Story, ingest);
-                    item.Language = GetFmsData(filtered, Fields.Lang, ingest);
-                    item.FilePath = document.Key;
-                    item.Summary = GetFmsData(filtered, Fields.Summary, ingest);
-                    item.Page = GetFmsData(filtered, Fields.Page, ingest);
-                    item.Section = GetFmsData(filtered, Fields.Section, ingest);
-
-                    item.Tags = GetTagList(filtered, ingest);
-                    item.Authors = GetAuthorList(GetFmsData(filtered, Fields.Author, ingest));
-                    item.PublishedOn = GetPublishedOn(GetFmsData(filtered, Fields.Date, ingest), ingest);
-
-                    articles.Add(item);
+                    code = string.IsNullOrEmpty(code) ? GetItemSourceCode(ingest, papername, sources) : code;
+
+                    if (!string.IsNullOrEmpty(code)) // This is a valid newspaper source
+                    {
+                        item.Source = code;
+                        item.Title = GetFmsData(filtered, Fields.Headline, ingest);
+                        item.Uid = GetFmsData(filtered, Fields.Id, ingest);
+                        item.Body = GetFmsData(preFiltered + "<break>", Fields.Story, ingest);
+                        item.FilePath = document.Key;
+                        item.Summary = GetFmsData(filtered, Fields.Summary, ingest);
+                        item.Page = GetFmsData(filtered, Fields.Page, ingest);
+                        item.Section = GetFmsData(filtered, Fields.Section, ingest);
+                        item.Language = ingest.GetConfigurationValue("language");
+                        item.ProductId = ingest.ProductId;
+
+                        item.Tags = GetTagList(filtered, ingest);
+                        item.Authors = GetAuthorList(GetFmsData(filtered, Fields.Author, ingest));
+                        item.PublishedOn = GetPublishedOn(GetFmsData(filtered, Fields.Date, ingest), ingest);
+
+                        articles.Add(item);
+                    }
                 }
             }
             catch (Exception ex)
```

Comment:
Doesn't seem right to log a warning and then throw the exception.

Reply:
See previous comment.